geotrellis-sentinelhub/src/main/scala/org/openeo/geotrellissentinelhub/PyramidFactory.scala

```diff
@@ -332,12 +332,12 @@ class PyramidFactory(collectionId: String, datasetId: String, catalogApi: Catalo
                 from, atEndOfDay(to), accessToken, Criteria.toQueryProperties(metadata_properties))
             }
 
-            tracker.addInputProducts(collectionId, features.keys.toList.asJava)
+            tracker.addInputProducts(collectionId, features.map(p => p._2.data.selfUrl.getOrElse(p._1)).toList.asJava)
```

Comment:
`features.map { case (id, Feature(_, FeatureData(_, selfUrl))) => selfUrl.getOrElse(id) }` is more readable than _1, _2 etc.

Reply:
Fixed in https://github.com/Open-EO/openeo-geotrellis-extensions/pull/134